### ISSUE TITLE
unexpected 307 redirect on firefox but not curl (based on UA)

### ISSUE THREAD

**Issue by danieljakots** [minio/minio#12537]
Hi,

I update from minio-0.20210527 to 0.20210616 and now I get unexpected redirect with firefox. I can reproduce the problem with curl if I set the UA to firefox.

```
$ curl -i 'https://dumpster.chown.me/mastodon/cache/accounts/avatars/000/000/536/original/6a375d0154af5184.png' -H 'Accept: image/webp,*/*' -H 'Accept-Language: en-CA,en-US;q=0.7,en;q=0.3' --compressed -H 'Referer: https://awoo.chown.me/' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'TE: Trailers' -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:89.0) Gecko/20100101 Firefox/89.0'
HTTP/2 307 
server: nginx
date: Sat, 19 Jun 2021 14:43:05 GMT
content-type: text/html; charset=utf-8
content-length: 66
location: http://dumpster.chown.me:32127
vary: Origin
strict-transport-security: max-age=15768000; includeSubDomains; preload;

<a href="http://dumpster.chown.me:32127">Temporary Redirect</a>.
````

The port is the one picked dynamically for the console endpoint (noticed because of the `WARNING: Console endpoint is listening on a dynamic port (32127), please use --console-address ":PORT" to choose a static port.` message.

If I comment out the UA setting in curl, it works fine:

```
$ curl -i 'https://dumpster.chown.me/mastodon/cache/accounts/avatars/000/000/536/original/6a375d0154af5184.png' -H 'Accept: image/webp,*/*' -H 'Accept-Language: en-CA,en-US;q=0.7,en;q=0.3' --compressed -H 'Referer: https://awoo.chown.me/' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'TE: Trailers' # -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:89.0) Gecko/20100101 Firefox/89.0'
HTTP/2 200 
server: nginx
date: Sat, 19 Jun 2021 14:43:10 GMT
content-type: image/png
content-length: 80549
accept-ranges: bytes
cache-control: public, max-age=315576000, immutable
content-security-policy: block-all-mixed-content
etag: "cc1d3fd2dede445dbb4314a46496c8bd"
last-modified: Mon, 17 May 2021 17:50:21 GMT
vary: Origin
x-amz-request-id: 168A02E33721EBAE
x-xss-protection: 1; mode=block
strict-transport-security: max-age=15768000; includeSubDomains; preload;

Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
```

I'm running minio with `$ /usr/local/bin/minio server /var/minio/export`. I don't get any log. I tried to set the console address like `$ /usr/local/bin/minio server /var/minio/export --console-address 9001` but it returns so I think minio dies.

I couldn't find how to enable logging with `$ /usr/local/bin/minio -h` :(

## Your Environment

```
$ /usr/local/bin/minio --version                       
minio version RELEASE.2021-06-16T23-19-26Z
```

Config is basic, I just want a S3 compatible server to do get and post. However the config is 794 lines, should I paste it anyway?

```
$ uname -a
OpenBSD web1.chown.me 6.9 GENERIC.MP#58 amd64
```

Cheers,
Daniel

**Comment by danieljakots** [comment]
I rolled back to the version I was running
```
$ /usr/local/bin/minio --version 
minio version RELEASE.2021-05-27T20-38-04Z
```
and it works again as intended.

**Comment by harshavardhana** [comment]
The redirects and change in browser code are only in master branch and not released how are you able to obtain versions that we haven't released?

If you are source compiling this is expected.

**Comment by harshavardhana** [comment]
There is no release made on the date you are referring to here 

> ```
> minio version RELEASE.2021-06-16T23-19-26Z
> ```

All our tags are for these releases.

RELEASE.2021-06-09T18-51-39Z
RELEASE.2021-06-14T01-29-23Z
RELEASE.2021-06-17T00-10-46Z

In case someone is building things locally and doing source compilation.. 

And yes if you a UA agent with "Mozilla" or "Firefox" in it we redirect to a browser UI. 

> The port is the one picked dynamically for the console endpoint (noticed because of the `WARNING: Console endpoint is listening on a dynamic port (32127), please use --console-address ":PORT" to choose a static port.` message.

This is only in master branch. Going to close this issue as custom builds. 

> I'm running minio with `$ /usr/local/bin/minio server /var/minio/export`. I don't get any log. I tried to set the console address like `$ /usr/local/bin/minio server /var/minio/export --console-address 9001` but it returns so I think minio dies.

You have to provide `--console-address :9001` the format is mentioned in the log `:PORT`

**Comment by danieljakots** [comment]
Indeed, it's a mistake on my side.

Regarding the version name, it's some difference between GH release name and golang release name. I was talking about RELEASE.2021-06-17T00-10-46Z 180eabaa8eb105bdfdf67f988d4bef3a8339074e. For some reasons, proxy.golang.org returns another timestamp for this commit:
https://proxy.golang.org/github.com/minio/minio/@v/v0.0.0-20210616231926-180eabaa8eb1.info 

```
{
  "Version": "v0.0.0-20210616231926-180eabaa8eb1",
  "Time": "2021-06-16T23:19:26Z"
}
```

(But that's not the problem here, I somehow ended with e1870c7b7c23 which is actually the last commit in master, like you said).

You can close, thanks!

**Comment by harshavardhana** [comment]
> Regarding the version name, it's some difference between GH release name and golang release name. I was talking about RELEASE.2021-06-17T00-10-46Z [180eaba](https://github.com/minio/minio/commit/180eabaa8eb105bdfdf67f988d4bef3a8339074e). For some reasons, proxy.golang.org returns another timestamp for this commit:

Yeah mainly because commit time is different than build time they do not need to match each other.

**Comment by harshavardhana** [comment]
FWIW adding another ENV to turn off browser redirects if your application doesn't like it such as `MINIO_BROWSER_REDIRECT=off`  available in PR #12534

**Comment by klausenbusk** [comment]
> FWIW adding another ENV to turn off browser redirects if your application doesn't like it such as `MINIO_BROWSER_REDIRECT=off` available in PR #12534

Was this removed on purpose in https://github.com/minio/minio/pull/12629 ?

**Comment by sudoexec** [comment]
> FWIW adding another ENV to turn off browser redirects if your application doesn't like it such as `MINIO_BROWSER_REDIRECT=off` available in PR #12534

It looks like that `MINIO_BROWSER_REDIRECT=off` doesn't work.
Will it be added back?

**Comment by Qbuer** [comment]
this [link](https://stackoverflow.com/questions/68317358/minio-local-dashboard-is-not-opening) may be useful for someone who encounter this problem.
